### PR TITLE
Update docs landing page

### PR DESCRIPTION
**Pull request by ccarterlandis**
Minor updates to the docs landing page.

### CODE REVIEW

File: docs/source/index.rst
Comment (sgoggins) [inline docs/source/index.rst:53]: Should probably acknowledge Trey. And we should probably call them "Current Core Contributors" and "Previous Core Contributors". .. And we should probably include our GSOC students. I would say we call everyone except Keanue a "Current Core Contributor", as even Parth is still doing a few things here and there.

**Review by sgoggins — APPROVED** [review]
I am approving this, providing my comment as a suggestion, but not a requested change. Ultimately I leave it to you, @gabe-heim and @howderek  (Who I think we almost hilariously omitted) to decide.

File: docs/source/index.rst
Comment (sgoggins) [reply docs/source/index.rst:53]: Also, @ChristianCme should be added to former core contributors, IMHO.

Comment (sgoggins) [reply docs/source/index.rst:53]: Oh, nevermind on @howderek ... he's at the top ... and I think @sgoggins and @germonprez should probably be at the bottom of the list if its not alphabetical. Other current core contributors are, I think @geekygirldawn and @ivanayov ....

**Review by sgoggins — APPROVED** [review]
Again, I approve. These are suggestions and corrections to my current comments.  As I go through the list, I also think we need to acknowledge the contributions of Remy and Brian Proffit. They didn't really contribute to the code base or issue stuff, but have been essential collaborators.

**Comment by ccarterlandis** [comment]
@sgoggins Respectfully, I think there is a non-trivial difference between a "core" contributor and a normal contributor. I don't mean to downplay the significance or impact of any contributor, but I do think that there is a certain level of sustained involvement & engagement that differentiates the two levels, and I don't think everyone you listed is a "core contributor". I would like to represent everyone's involvement in the project fairly, and I think those of us that have work as part of Augur's active core development team should be listed in a separate section than the less frequent contributors. I'm not saying that contributing less frequently is undesirable, but the core contributors/maintainers are the primary project contacts and are the ones to go to with questions or issues, so we should list ourselves as such. The more people we mark as "core" contributors, the less meaning it actually has, because we would be using the word "core" so broadly. 

I think it would be more appropriate to have a section for the current and past maintainers, and then possibly other contributors, and then perhaps GSoC participants. Something like:

### Current maintainers
You, me, Matt G., Gabe, Derek, Matt S., Ryland, Andrew, Max(?), 

### Former maintainers
Carolyn, Elita, Jonah, Michael, Christian

### Other contributors
Remy, Brian, Ivana, Dawn, Georg, etc.

### GSoC 2020
Akshara, Tianyi, Pratik, Sarit, Sai, Abhinav

### GSoC 2019
Parth, Bingwen

### GSoC 2018
Keanu

In all honesty, we might not even need the other contributors section unless we define a strict criteria for being added to it, since [the "Contributors" page](https://github.com/chaoss/augur/graphs/contributors) already shows everyone who's ever submitted a patch to the project.

File: docs/source/index.rst
Comment (ccarterlandis) [reply docs/source/index.rst:53]: There is no set order

**Comment by sgoggins** [comment]
@ccarterlandis : That's a good suggestion. I support it.

**Comment by ccarterlandis** [comment]
@sgoggins this was not supposed to be merged, I have not implemented the changes I suggested yet...

**Pull request merged**
